lib/streamlit/delta_generator.py

```diff
@@ -641,11 +639,13 @@ def _block(
 
     def _legacy_add_rows(
         self: DG,
-        data: "Data" = None,
-        **kwargs: Union[
-            "DataFrame", "npt.NDArray[Any]", Iterable[Any], Dict[Hashable, Any], None
-        ],
-    ) -> Optional[DG]:
+        data: Data = None,
+        **kwargs: DataFrame
+        | npt.NDArray[Any]
```

Comment:
I'm not 100% sure, but I think there was some kind of problem with older numpy versions if `NDArray` is used that way. The reason is that `NDArray` was only added a few versions ago.